### ISSUE TITLE
Show invalid column names correctly

### ISSUE THREAD

**Issue by jamcoupe** [sqlfluff/sqlfluff#5287]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

When I define the following table (sqlite):

```sql
CREATE TABLE IF NOT EXISTS testing (
    end DATETIME NOT NULL
);
```

I get an unparsable error of:
```
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE                                                                                                                           
                       | TABLE IF NOT EXISTS testing (\n   ...'
```

### Use case

What would be nice is if the linter told me that the error was due to using an invalid column name and/or suggest to wrap it in quotes.

### Dialect

sqlite

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by joaostorrer** [comment]
The parsing error is because `end` is a reserved keyword in SQLite.
If you quote the column name it parses correctly:

```
CREATE TABLE IF NOT EXISTS testing (
    "end" DATETIME NOT NULL
);
```

**Comment by bressanmarcos** [comment]
I got the exact same message when calling a column DELTA when using the snowflake dialect. 

Quoting the word worked, but the error message should definitely be improved

**Comment by peterbud** [comment]
Hi @jamcoupe 

Thank you for opening this issue and sharing the use case!

I completely understand how a more specific error message—like identifying `end` as a reserved keyword—would improve the developer experience. However, there is a technical constraint in how the linter's engine handles these cases.

When the parser encounters a syntax error (like using the reserved word `end` without quotes), it reaches a state where the rest of the statement no longer matches any valid SQL patterns. Because the linter can no longer "read" the code past that point, it marks the entire block as an unparsable section.

In short:

* Because the parser "breaks" at the unexpected token, it loses the context necessary to determine why it failed (e.g., distinguishing between a typo, a missing comma, or a reserved keyword).

* Currently, we can only identify where the breakdown occurred, rather than the specific semantic cause.

That said, we appreciate the feedback. While we can't easily implement "smart" suggestions for unparsable blocks right now, we’re always looking for ways to improve our grammar definitions to make these errors less frequent.

For now, wrapping reserved words in quotes (e.g., "end") is the best way to resolve this for the linter.
